zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/factory/SharedZkClientFactory.java

```diff
@@ -128,4 +139,78 @@ public int getActiveConnectionCount() {
     }
     return count;
   }
+
+  public interface OnCloseCallback {
+    /**
+     * Triggered after the SharedZkClient is closed.
+     */
+    void onClose();
+  }
+
+  /**
+   * NOTE: do NOT use this class directly. Please use SharedZkClientFactory to create an instance of SharedZkClient.
+   * InnerSharedZkClient is a ZkClient used by SharedZkClient to power ZK operations against a single ZK realm.
+   *
+   * NOTE2: current InnerSharedZkClient replace the original SharedZKClient. We intend to keep the behavior of original
+   * SharedZkClient intact. (Think of rename the original SharedZkClient as InnerSharedZkClient. This would maintain
+   * backward compatibility.
+   */
+  public static class InnerSharedZkClient extends ZkClient implements HelixZkClient {
```

Comment:
Is this moved from original SharedZKClient?

Reply:
Yes, you are right. The intention is to keep the InnerSharedZkClient exactly the same as original sharedZkClient. It is just we need the name of SharedZkClient that we renamed InnerSharedZkClient.